src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -184,13 +184,15 @@ private static IExpression CreateParamArray(IParameterSymbol parameter, Immutabl
             if (parameter.Type.TypeKind == TypeKind.Array)
             {
                 IArrayTypeSymbol arrayType = (IArrayTypeSymbol)parameter.Type;
-                ArrayBuilder<IExpression> paramArrayArguments = ArrayBuilder<IExpression>.GetInstance(boundArguments.Length - firstArgumentElementIndex);
+                ArrayBuilder<IExpression> builder = ArrayBuilder<IExpression>.GetInstance(boundArguments.Length - firstArgumentElementIndex);
+
                 for (int index = firstArgumentElementIndex; index < boundArguments.Length; index++)
                 {
-                    paramArrayArguments.Add(boundArguments[index]);
+                    builder.Add(boundArguments[index]);
                 }
+                var paramArrayArguments = builder.ToImmutableAndFree();
 
-                return new ArrayCreation(arrayType, paramArrayArguments.ToImmutableAndFree(), boundArguments.Length - 1 > firstArgumentElementIndex ? boundArguments[firstArgumentElementIndex].Syntax : null);
+                return new ArrayCreation(arrayType, paramArrayArguments, paramArrayArguments.Length > 0 ? paramArrayArguments[0].Syntax : null);
```

Comment:
If I understand Gen correctly - if there's code like `Foo(regularParam, paramsParam1, paramsParam2)` the syntaxtree will only have one SyntaxList for all the arguments and so for this operation what would be the syntax to return - paramsParam1's syntax or a conjured up SyntaxList(paramsParam1, paramsParam2)? 

I don't know if we conjure up new syntax node anywhere for the operation trees - if we don't I wouldn't start doing that now.